ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -70,6 +70,34 @@ type domain_info = {
         (* Persist netbios_name to support hostname change *)
 }
 
+module KerberosEncryptionTypes = struct
+  type t = Strong | Legacy | All
+
+  let default = Strong
+
+  let to_string = function
+    | Strong ->
+        "strong"
+    | Legacy ->
+        "legacy"
+    | All ->
+        "all"
+
+  let of_string = function
+    | "all" ->
+        All
+    | "legacy" ->
+        Legacy
+    | "strong" ->
+        Strong
+    | _ as v ->
+        debug "Invalid kerberos type %s use default %s" v (to_string default) ;
+        default
+
+  let get () =
+    !Xapi_globs.winbind_kerberos_encryption_type |> of_string |> to_string
```

Comment:
This should return `t` or otherwise it does not make much sense.

Reply:
why it should return `t`, 
I want to use `of_string` and `to_string` to check the string value validation and provide default value if not, does not make sense?

Reply:
> why it should return t,

The client that needs to check properties of the value will be forced to parse again the string. I prefer the interface to return `t` and the user convert to string if needed.